### DISCUSSION TITLE
Error validating repository information. Credentials ok.

### DISCUSSION THREAD

**Discussion by advitramesh** [community/community#77799] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hello Experts 

I am using Jenkins as a build server to sync my code to an organization repo. I am using credentials of kind GitHub App.

In my pipeline job I get an error "Error validating repository information. Credentials ok. "when I click on validate .  The detailed error description is mentioned in the attachment 
[error_Jenkins_pipeline_1.o.txt](https://github.com/community/community/files/13540171/error_Jenkins_pipeline_1.o.txt)

I have read some answers on the community and I take that I have to create a Personal Access Token. Can I correct the above issue using my GitHub App. itself? 

Please note the list of permissions for my gitHub App 

[permissions.txt](https://github.com/community/community/files/13540185/permissions.txt)

Thanks for your time 
Regards
Advit

**Comment by phattharaphorn** [comment, marked as answer]
Hello Advit,

It seems like you're facing an issue with Jenkins and GitHub App credentials while syncing your code. The error message "Error validating repository information. Credentials ok." can be a bit tricky, and there might be several reasons for it.

Firstly, it's important to note that GitHub Apps generally don't use Personal Access Tokens (PATs). They have their own installation tokens that are used to authenticate requests. So, creating a PAT might not be the solution.
Here are a few things you can check and try:

- Check GitHub App Permissions:
Ensure that your GitHub App has the necessary permissions for repository access. The permissions.txt file you mentioned could be crucial in verifying this.

- Review Jenkins Configuration:
Double-check the Jenkins configuration for your GitHub App credentials. Make sure you're using the correct App ID, Private Key, and other relevant details.

- GitHub App Installation:
Verify that the GitHub App is installed correctly on the organization/repository you are trying to sync.

- Check Webhook Configurations:
Ensure that the webhook for your GitHub App is properly configured. It's required for Jenkins to receive events from GitHub.

- Jenkins Plugin Version:
Make sure you are using the latest version of the GitHub plugin for Jenkins. Sometimes, compatibility issues can cause authentication problems.

- Logs and Debugging:
Enable debug logging in Jenkins to get more detailed information about the error. This might provide additional insights into what is going wrong.

Without the specific details from the error_Jenkins_pipeline_1.o.txt file, it's challenging to provide a more targeted solution. If you can share relevant portions of the error log, I might be able to assist you more effectively.

Remember to avoid sharing sensitive information such as private keys or tokens in public forums. If needed, you can redact or replace sensitive information before sharing logs.

**Reply by advitramesh** [reply]
Thanks 

The issue was with the repository we have linked was not granted access to the GitHub App installation
